actions/format_image_action.go

```diff
@@ -0,0 +1,157 @@
+/*
+FormatImage Action
+
+This action creates an image file and formats it with a filesystem.
+
+Yaml syntax:
+ - action: format-image
+   imagename: image_name
+   imagesize: size
+   fs: filesystem
+   blocksize: 4096
+
+Mandatory properties:
+
+- imagename -- the name of the image file.
+
+- imagesize -- generated image size in human-readable form, examples: 100MB, 1GB, etc.
+
+- fs -- filesystem type used for formatting.
+
+Optional properties:
+
+- blocksize -- size of blocks in bytes, valid values are 1024, 2048 and 4096. For ext fs only.
+*/
+
+package actions
+
+import (
+	"fmt"
+	"github.com/docker/go-units"
+	"github.com/go-debos/fakemachine"
+	"os"
+	"path"
+	"strings"
+	"syscall"
+
+	"github.com/go-debos/debos"
+)
+
+type FormatImageAction struct {
+	debos.BaseAction `yaml:",inline"`
+	ImageName        string
+	ImageSize        string
+	FS               string
+	BlockSize        int
+	size             int64
+}
+
+func (i FormatImageAction) format(context debos.DebosContext) error {
```

Comment:
You'll probably want to factor this out form image-partition instead so that we don't have two copies of the code doing the same thing.